docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+* Depth of the word in the synonym net
+
+The following example defines a synonym set for the words "come", "advance" and "approach":
+
+[source,synonyms]
+----
+s(100000002,1,'come',v,1,0).
+s(100000002,2,'advance',v,1,0).
+s(100000002,3,'approach',v,1,0).""";
+----
+
+[discrete]
+[[synonyms-store-synonyms]]
+=== Store your synonyms set
+
+Your synonyms sets need to be stored in {es} so your analyzers can refer to them.
+There are three ways to include your synonyms sets:
+
+[discrete]
+[[synonyms-store-synonyms-api]]
+==== Synonyms API
+
+You can use the <<synonyms-apis,synonyms APIs>> to manage synonyms sets.
+This is the most flexible approach, as it allows to dynamically define and modify synonyms sets.
+
+Changes in your synonyms sets will automatically reload the associated analyzers.
+
+[discrete]
+[[synonyms-store-synonyms-file]]
+==== Synonyms File
+
+You can store your synonyms set in a file.
+
+A synonyms set file needs to be uploaded to all your cluster nodes, and be located in the configuration directory for your {es} distribution.
+If you're using {ess}, you can upload synonyms files using {cloud}/ec-custom-bundles.html[custom bundles].
+
+An example synonyms file:
+
+[source,synonyms]
+--------------------------------------------------
+include::{es-test-dir}/cluster/config/analysis/synonym.txt[]
+--------------------------------------------------
+
+To update an existing synonyms set, upload new files to your cluster.
+Synonyms set files must be kept in sync on every cluster node.
+
+When a synonyms set is updated, search analyzers that use it need to be refreshed using the <<indices-reload-analyzers,reload search analyzers API>>
+
+This manual syncing and reloading makes this approach less flexible than using the <<synonyms-store-synonyms-api,synonyms API>>.
+
+[discrete]
+[[synonyms-store-synonyms-inline]]
+==== Inline
+
+For testing out your synonyms, you can directly inline them in your token filter definition.
+
+[WARNING]
+======
+This is not recommended for production usage.
+A large number of inline synonyms increases cluster size unnecessarily and can lead to performance issues.
+======
+
+[discrete]
+[[synonyms-synonym-analyzers]]
+=== Configure your analyzers to use synonyms sets
+
+[discrete]
+[[synonyms-synonym-token-filters]]
+==== Configure synonyms token filters
+
+Once your synonyms sets are created, you can start configuring your token filters and analyzers to use them.
+
+{es} uses synonyms as part of the <<analysis-overview,analysis process>>.
+There are two different <<analysis-tokenfilters,token filters>> that can be used for including synonyms:
+
+* <<analysis-synonym-tokenfilter>>: Use it for single word synonyms ("car", "automobile").
+* <<analysis-synonym-graph-tokenfilter>>: Use it for synonyms that may span multiple words ("hurriedly", "in a hurry").
+
+Synonyms sets created via synonyms APIs can be included into synonym token filters using `synonyms_set` configuration:
+
```

Comment:
why "can be included"?
May be we should say like this:

"Use "synonyms_set" option to provide a synonym set created via Synonyms Management APIs"

"Use "synonyms_path" option to provide a synonym file.

"Use "synonyms" option to list inline synonyms.

Reply:
Sounds more direct, thanks!